commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/SumTest.java

```diff
@@ -0,0 +1,176 @@
+            Arguments.of(new double[] {Double.POSITIVE_INFINITY, Double.MAX_VALUE},
+                Double.POSITIVE_INFINITY),
+            Arguments.of(new double[] {Double.MAX_VALUE, Double.MAX_VALUE},
+                Double.POSITIVE_INFINITY),
+            Arguments.of(new double[] {-Double.MAX_VALUE, Double.POSITIVE_INFINITY},
+                Double.POSITIVE_INFINITY),
+            Arguments.of(new double[] {-Double.MAX_VALUE, -Double.MAX_VALUE},
+                Double.NEGATIVE_INFINITY),
+            Arguments.of(new double[] {Double.NEGATIVE_INFINITY, -Double.MIN_VALUE},
+                Double.NEGATIVE_INFINITY),
+            Arguments.of(new double[] {Double.NaN, 34.56, 89.74}, Double.NaN),
+            Arguments.of(new double[] {34.56, Double.NaN, 89.74}, Double.NaN),
+            Arguments.of(new double[] {34.56, 89.74, Double.NaN}, Double.NaN),
+            Arguments.of(new double[] {Double.NaN, 3.14, Double.NaN, Double.NaN}, Double.NaN),
+            Arguments.of(new double[] {Double.NaN, Double.NaN, Double.NaN}, Double.NaN),
+            Arguments.of(new double[] {Double.NEGATIVE_INFINITY, Double.MAX_VALUE},
+                Double.NEGATIVE_INFINITY),
+            Arguments.of(new double[] {Double.MAX_VALUE, 1}, Double.MAX_VALUE),
+            Arguments.of(new double[] {-Double.MAX_VALUE, 1, 1}, -Double.MAX_VALUE),
+            Arguments.of(new double[] {-Double.MAX_VALUE, -1, 1}, -Double.MAX_VALUE),
+            Arguments.of(new double[] {Double.MAX_VALUE, -1}, Double.MAX_VALUE),
+            Arguments.of(new double[] {-Double.MAX_VALUE, 1, -1}, -Double.MAX_VALUE),
+            Arguments.of(new double[] {1, 2, 3, Double.MAX_VALUE, -Double.MAX_VALUE, -7}, -1)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "testSum")
+    void testParallelStream(double[] values, double expected) {
+        double ans = Arrays.stream(values)
+                .parallel()
+                .collect(Sum::create, Sum::accept, Sum::combine)
+                .getAsDouble();
+        Assertions.assertEquals(expected, ans);
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "testSum")
+    void testSumRandomOrder(double[] values, double expected) {
+        UniformRandomProvider rng = TestHelper.createRNG();
+        for (int i = 0; i < 10; i++) {
+            testSum(TestHelper.shuffle(rng, values), expected);
+            testParallelStream(TestHelper.shuffle(rng, values), expected);
+            testCombine(TestHelper.shuffle(rng, values), expected);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "testSum")
+    void testCombine(double[] values, double expected) {
+        Sum sum1 = Sum.create();
+        Sum sum2 = Sum.create();
+        Arrays.stream(values).forEach(sum1);
+        Arrays.stream(values).forEach(sum2);
+        double sum2BeforeCombine = sum2.getAsDouble();
+        sum1.combine(sum2);
+        Assertions.assertEquals(expected * 2, sum1.getAsDouble(), 10E-12);
```

Comment:
The delta here may not always be suitable since it is absolute and not relative. For a quick relative test you can use:
```java
double eps = 1e-15; // for example
assertEquals(expected, actual, Math.abs(expected) * eps)
```
If you only expect a tiny difference then we can use Precision.equals with a ulp argument. This may be more relevant here.

Since you lose the error message containing the values from the assertion with doubles you have to provide it. As such this is useful to put into a method in the TestHelper class:
```java
static void assertEquals(double expected, double actual, int ulps, Supplier<String> msg) {
    Assertions.assertTrue(Precision.equals(expected, actual, ulps),
        () -> equals + " != " + actual + " within " + ulps + " ulps: " + msg.get());
}
```
Note: This is untested and could be improved to create an error message more consistent with JUnit (fixing the error message is not a priority).